app/services/katello/pulp3/repository.rb

```diff
@@ -111,10 +117,28 @@ def delete_remote(href = repo.remote_href)
         remotes_api.delete(href) if href
       end
 
-      def list_remotes(args)
+      def self.delete_remote(smart_proxy, href)
+        remotes_api(smart_proxy).delete(href) if href
+      end
+
+      def remotes_list(args)
         remotes_api.list(args).results
       end
 
+      def self.delete_orphan_remotes(smart_proxy, repo_types = pulp3_enabled_repo_types(smart_proxy))
+        tasks = []
+        repo_types.each do |repo_type|
+          pulp3_class = repo_type.pulp3_service_class
+          remotes = pulp3_class.remotes_list(smart_proxy, {})
+          repos_on_capsule = ::Katello::Pulp3::Repository.new(nil, smart_proxy).list({})
+          pulp_ids = repos_on_capsule.collect { |repo| repo.name }
+          remotes.each do |remote|
+            tasks << pulp3_class.delete_remote(smart_proxy, remote['pulp_href']) unless pulp_ids.include?(remote['name]'])
+          end
```

Comment:
can you add a comment explaining how this is working, it took me a a minute to figure out  how or why its working (the fact its comparing repository name to remote name).  Also, renaming pulp_ids to repository_name might make it more clear

Reply:
*repository_names

Reply:
I think the naming of `pulp_ids` is a carryover from other operations where the pulp fields are actually pulp ids, but here it really doesn't make sense. It really should be something like `remote_name`, etc. Sorry for the confusion.

Reply:
also, when testing this, it seemed to delete all my remotes, even though it should have left 1?  Are you seeing it working properly?